### PR TITLE
Forms: Introduce form field

### PR DESCRIPTION
**Pull request by dprokop**
Ref #19855

Introduces `Forms.Field` component to handle label/description/form input/validation message combo. See documentation in code/story for information about API

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/Field.story.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    </div>
+  );
+};
+
+export const horizontalLayout = () => {
+  const { containerWidth, ...otherProps } = getKnobs();
+  return (
+    <div style={{ width: containerWidth }}>
+      <Field horizontal label="Show labels" description="Display thresholds's labels" {...otherProps}>
+        <Switch />
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Field.story.tsx]: Could this be a working example? The switch does not work for me in storybook

File: packages/grafana-ui/src/components/Forms/Field.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  disabled,
+  error,
+  children,
+  className,
+}) => {
+  const theme = useTheme();
+  let inputId;
+  const styles = getFieldStyles(theme);
+
+  // Get the firs, and only, child to retrieve form input's id
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Field.tsx]:
```suggestion
  // Get the first, and only, child to retrieve form input's id
```

File: packages/grafana-ui/src/components/Forms/Label.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { useTheme, stylesFactory } from '../../themes';
 import { GrafanaTheme } from '@grafana/data';
 import { css, cx } from 'emotion';
 
-export interface LabelProps extends React.HTMLAttributes<HTMLLabelElement> {
+export interface LabelProps extends React.LabelHTMLAttributes<HTMLLabelElement> {
```
Comment (peterholmberg) [inline packages/grafana-ui/src/components/Forms/Label.tsx]: What's the difference between these two?

**Review by peterholmberg — CHANGES_REQUESTED** [review]

File: packages/grafana-ui/src/components/Forms/Label.tsx (hunk 3 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Forms/Label.tsx]: `LabelHTMLAttributes` has `htmlFor` prop on it

File: packages/grafana-ui/src/components/Forms/Field.story.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Forms/Field.story.tsx]: done

**Comment by torkelo** [comment]
This looks great! Really like the children approach vs inputEl

**Review by peterholmberg — APPROVED** [review]
LGTM

**Pull request merged**